### ISSUE TITLE
Panic when using docker_env_metadata_whitelist

### ISSUE THREAD

**Issue by deedubs** [google/cadvisor#1078]
I've updated to 0.20.4 to get docker_env_metadata_whitelist but I'm getting a panic

> openssl sha1 /usr/local/bin/cadvisor
SHA1(/usr/local/bin/cadvisor)= 437fad6ce65dc4fdb162b5d72f6d3aa1e7d5956f
> sudo /usr/local/bin/cadvisor -port=8082 -event_storage_age_limit=1m -docker_only=true -docker_env_metadata_whitelist=MARATHON_APP_ID,MARATHON_APP_DOCKER_IMAGE

````
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/google/cadvisor/container/docker.newDockerContainerHandler(0xc820170360, 0xc8200a2a50, 0x48, 0x7f91e1ea67d0, 0xc8200dea80, 0x7f91e1ea5f10, 0xc820248e00, 0xc8202d5cf7, 0x4, 0xc820380628, ...)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/container/docker/handler.go:163 +0x11f2
github.com/google/cadvisor/container/docker.(*dockerFactory).NewContainerHandler(0xc820380600, 0xc8200a2a50, 0x48, 0xc8205dd501, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/container/docker/factory.go:135 +0x17e
github.com/google/cadvisor/container.NewContainerHandler(0xc8200a2a50, 0x48, 0xc8205dd801, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/container/factory.go:79 +0x6e6
github.com/google/cadvisor/manager.(*manager).createContainer(0xc8200dea80, 0xc8200a2a50, 0x48, 0x0, 0x0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:761 +0x170
github.com/google/cadvisor/manager.(*manager).detectSubcontainers(0xc8200dea80, 0xc141a0, 0x1, 0x0, 0x0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:926 +0x1e9
github.com/google/cadvisor/manager.(*manager).Start(0xc8200dea80, 0x0, 0x0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:253 +0x60f
main.main()
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/cadvisor.go:99 +0xa4c

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 7 [sleep]:
time.Sleep(0x54d05953)
	/usr/local/go/src/runtime/time.go:59 +0xf9
github.com/google/cadvisor/manager.(*containerData).housekeeping(0xc8200691e0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/container.go:438 +0x80e
created by github.com/google/cadvisor/manager.(*containerData).Start
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/container.go:94 +0x41

goroutine 18 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 19 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x13244a0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:882 +0x67
created by github.com/golang/glog.init.1
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x297

goroutine 30 [syscall]:
syscall.Syscall(0x0, 0x7, 0xc8203c9e20, 0x10000, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x7, 0xc8203c9e20, 0x10000, 0x10000, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x5f
syscall.Read(0x7, 0xc8203c9e20, 0x10000, 0x10000, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
golang.org/x/exp/inotify.(*Watcher).readEvents(0xc820291800)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/Godeps/_workspace/src/golang.org/x/exp/inotify/inotify_linux.go:156 +0xab
created by golang.org/x/exp/inotify.NewWatcher
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/Godeps/_workspace/src/golang.org/x/exp/inotify/inotify_linux.go:76 +0x365

goroutine 38 [IO wait]:
net.runtime_pollWait(0x7f91e062c970, 0x72, 0xc820016180)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201d8bc0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201d8bc0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201d8b60, 0xc82022a000, 0x1000, 0x1000, 0x0, 0x7f91e44f2050, 0xc820016180)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc82015a270, 0xc82022a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
net/http.noteEOFReader.Read(0x7f91e062ca30, 0xc82015a270, 0xc82019e3c8, 0xc82022a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc8201c1be0, 0xc82022a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc8201736e0)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc8201736e0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc82019e370)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 39 [select]:
net/http.(*persistConn).writeLoop(0xc82019e370)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 42 [sleep]:
time.Sleep(0x5f5e100)
	/usr/local/go/src/runtime/time.go:59 +0xf9
github.com/google/cadvisor/utils/oomparser.readLinesFromFile(0xc820173c80, 0xc8203970e0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/utils/oomparser/oomparser.go:118 +0x14b
github.com/google/cadvisor/utils/oomparser.(*OomParser).StreamOoms.func1(0xc820173c80, 0xc8200f46e0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/utils/oomparser/oomparser.go:138 +0x2e
created by github.com/google/cadvisor/utils/oomparser.(*OomParser).StreamOoms
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/utils/oomparser/oomparser.go:139 +0x78

goroutine 31 [chan receive]:
github.com/google/cadvisor/utils/oomparser.(*OomParser).StreamOoms(0xc8200f46e0, 0xc820397080)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/utils/oomparser/oomparser.go:157 +0x393
created by github.com/google/cadvisor/manager.(*manager).watchForNewOoms
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:1006 +0xc9

goroutine 32 [chan receive]:
github.com/google/cadvisor/manager.(*manager).watchForNewOoms.func1(0xc820397080, 0xc8200dea80)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:1009 +0x6b
created by github.com/google/cadvisor/manager.(*manager).watchForNewOoms
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:1038 +0xf5

goroutine 52 [sleep]:
time.Sleep(0x64858785)
	/usr/local/go/src/runtime/time.go:59 +0xf9
github.com/google/cadvisor/manager.(*containerData).housekeeping(0xc820168680)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/container.go:438 +0x80e
created by github.com/google/cadvisor/manager.(*containerData).Start
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/container.go:94 +0x41

goroutine 53 [select]:
github.com/google/cadvisor/container/docker.(*realFsHandler).trackUsage(0xc8205ea300)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/container/docker/fsHandler.go:90 +0x53d
created by github.com/google/cadvisor/container/docker.(*realFsHandler).start
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/container/docker/fsHandler.go:107 +0x35
deedubs@thrust:~$ /usr/local/bin/cadvisor -port=8081 -event_storage_age_limit=1m -docker_only=true -docker_env_metadata_whitelist=MARATHON_APP_ID,MARATHON_APP_DOCKER_IMAGE/usr/local/bin/cadvisor -port=8081 -event_storage_age_limit=1m -docker_only=true -docker_env_metadata_whitelist=MARATHON_APP_ID,MARATHON_APP_DOCKER_IMAGE/usr/local/bin/cadvisor -port=8081 -event_storage_age_limit=1m -docker_only=true -docker_env_metadata_whitelist=MARATHON_APP_ID,MARATHON_APP_DOCKER_IMAGE^C
deedubs@thrust:~$ ^C
deedubs@thrust:~$ ^C
deedubs@thrust:~$ ^C
deedubs@thrust:~$ ^C
deedubs@thrust:~$ ^C
deedubs@thrust:~$ ^C
deedubs@thrust:~$ sudo /usr/local/bin/cadvisor -port=8082 -event_storage_age_limit=1m -docker_only=true -docker_env_metadata_whitelist=MARATHON_APP_ID,MARATHON_APP_DOCKER_IMAGE
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/google/cadvisor/container/docker.newDockerContainerHandler(0xc820170360, 0xc8203ddbd0, 0x48, 0x7f52d42a8968, 0xc8201bc780, 0x7f52d405ec80, 0xc8202554e0, 0xc820317bc7, 0x4, 0xc820375828, ...)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/container/docker/handler.go:163 +0x11f2
github.com/google/cadvisor/container/docker.(*dockerFactory).NewContainerHandler(0xc820375800, 0xc8203ddbd0, 0x48, 0xc820131501, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/container/docker/factory.go:135 +0x17e
github.com/google/cadvisor/container.NewContainerHandler(0xc8203ddbd0, 0x48, 0xc820131801, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/container/factory.go:79 +0x6e6
github.com/google/cadvisor/manager.(*manager).createContainer(0xc8201bc780, 0xc8203ddbd0, 0x48, 0x0, 0x0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:761 +0x170
github.com/google/cadvisor/manager.(*manager).detectSubcontainers(0xc8201bc780, 0xc141a0, 0x1, 0x0, 0x0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:926 +0x1e9
github.com/google/cadvisor/manager.(*manager).Start(0xc8201bc780, 0x0, 0x0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:253 +0x60f
main.main()
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/cadvisor.go:99 +0xa4c

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 18 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 19 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x13244a0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:882 +0x67
created by github.com/golang/glog.init.1
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x297

goroutine 26 [syscall]:
syscall.Syscall(0x0, 0x7, 0xc8203cbe20, 0x10000, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x7, 0xc8203cbe20, 0x10000, 0x10000, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x5f
syscall.Read(0x7, 0xc8203cbe20, 0x10000, 0x10000, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
golang.org/x/exp/inotify.(*Watcher).readEvents(0xc82014fe00)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/Godeps/_workspace/src/golang.org/x/exp/inotify/inotify_linux.go:156 +0xab
created by golang.org/x/exp/inotify.NewWatcher
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/Godeps/_workspace/src/golang.org/x/exp/inotify/inotify_linux.go:76 +0x365

goroutine 39 [IO wait]:
net.runtime_pollWait(0x7f52d405e870, 0x72, 0xc820014180)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201e0ae0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201e0ae0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201e0a80, 0xc82022a000, 0x1000, 0x1000, 0x0, 0x7f52d68f4050, 0xc820014180)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8201221b8, 0xc82022a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
net/http.noteEOFReader.Read(0x7f52d405e930, 0xc8201221b8, 0xc82019c3c8, 0xc82022a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc8201c98c0, 0xc82022a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc82011d680)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc82011d680, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc82019c370)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 40 [select]:
net/http.(*persistConn).writeLoop(0xc82019c370)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 27 [chan receive]:
github.com/google/cadvisor/utils/oomparser.(*OomParser).StreamOoms(0xc8200ee4b0, 0xc8203a22a0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/utils/oomparser/oomparser.go:157 +0x393
created by github.com/google/cadvisor/manager.(*manager).watchForNewOoms
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:1006 +0xc9

goroutine 28 [chan receive]:
github.com/google/cadvisor/manager.(*manager).watchForNewOoms.func1(0xc8203a22a0, 0xc8201bc780)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:1009 +0x6b
created by github.com/google/cadvisor/manager.(*manager).watchForNewOoms
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/manager.go:1038 +0xf5

goroutine 6 [sleep]:
time.Sleep(0x5f5e100)
	/usr/local/go/src/runtime/time.go:59 +0xf9
github.com/google/cadvisor/utils/oomparser.readLinesFromFile(0xc82001a5a0, 0xc8203a2300)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/utils/oomparser/oomparser.go:118 +0x14b
github.com/google/cadvisor/utils/oomparser.(*OomParser).StreamOoms.func1(0xc82001a5a0, 0xc8200ee4b0)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/utils/oomparser/oomparser.go:138 +0x2e
created by github.com/google/cadvisor/utils/oomparser.(*OomParser).StreamOoms
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/utils/oomparser/oomparser.go:139 +0x78

goroutine 7 [sleep]:
time.Sleep(0x54d8843d)
	/usr/local/go/src/runtime/time.go:59 +0xf9
github.com/google/cadvisor/manager.(*containerData).housekeeping(0xc820069520)
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/container.go:438 +0x80e
created by github.com/google/cadvisor/manager.(*containerData).Start
	/usr/local/google/home/vishnuk/go/src/github.com/google/cadvisor/manager/container.go:94 +0x41
````

**Comment by rikwasmus** [comment]
Running into it as well. Pull request is already there: https://github.com/google/cadvisor/pull/1063

**Comment by jimmidyson** [comment]
Yeah looking at merging now.

**Comment by jimmidyson** [comment]
Fixed by #1063
